### PR TITLE
implement crosshair

### LINKED ISSUE #409: [feature request] crosshair
This is another regular feature request. If we don't implement the projected crosshair from Crispy (we won't), this could be a rather straightforward porting over. It's still up for decision which patch to use for the actual rendering or if we should offer a selection of patches at all. In Boom/MBF we have another (smaller) HUD font to choose from. 

This pulls in quite a handful of additional config keys and will require additional care not to trigger INTERCEPT overflow in Vanilla complevel. 

This requires translucent and color translated patch drawing functions as a prerequisite.

### PR DESCRIPTION
**Pull request by rfomin**
This is a very simple implementation. Do I have to hardcode the crosshair patch? Or should we move our hardcoded data to wads?

Fixes #409

### CODE REVIEW

File: Source/hu_stuff.c [hunk 1]
```diff
@@ -823,6 +825,30 @@ static void HU_widget_build_sttime(void)
...
+    static patch_t *patch = NULL;
+
+    if (plr->playerstate != PST_LIVE ||
+        automapactive ||
+        menuactive ||
+        paused ||
+        secret_on)
+        return;
+
+    if (!patch)
```
Comment (fabiangreffrath) [inline Source/hu_stuff.c:839]: We should catch the case that the patch is not available.

**Comment by fabiangreffrath** [comment]
This is fine, but I'd like to add
 * three different crosshair shapes
 * color based on player health 
 * highlight on shootable target

**Comment by fabiangreffrath** [comment]
@rfomin Are you still working on this?

**Comment by rfomin** [comment]
> @rfomin Are you still working on this?

I've got distracted, sorry. Not very motivated to work on this, I don't like this feature. I will finish it eventually.

**Comment by fabiangreffrath** [comment]
I can take it over if you like.

**Comment by rfomin** [comment]
> I can take it over if you like.

Oh sure.

**Pull request closed without merging**